### PR TITLE
fix cmake find .h file by mistake

### PR DESCRIPTION
**Pull request by kuizhiqing**
### PR types

Bug fixes

### PR changes

Others

### Describe

FIND will find not only .h file but also the file contains .h in the path.

### CODE REVIEW

File: cmake/pten.cmake [hunk 1]
```diff
@@ -20,8 +20,8 @@ string(FIND ${TARGET_PATH} "experimental" pos)
 if (pos GREATER 1)
     file(GLOB HEADERS "${TARGET_PATH}/*" "*.h")
     foreach(header ${HEADERS})
-        string(FIND ${header} ".h" hpos)
-        if (hpos GREATER 1)
+        string(FIND ${header} ".h" hpos REVERSE)
+        if (hpos EQUAL 1)
```
Comment (chenwhql) [inline cmake/pten.cmake:24]: why use EQUAL?

File: cmake/pten.cmake [hunk 2]
```diff
@@ -40,4 +40,4 @@ pten_header_path_compat(${PADDLE_INFERENCE_INSTALL_DIR}/paddle/include/experimen
 
 # In order to be compatible with the original behavior, the header file name needs to be changed
 file(RENAME ${PADDLE_INFERENCE_INSTALL_DIR}/paddle/include/experimental/extension.h
-            ${PADDLE_INFERENCE_INSTALL_DIR}/paddle/include/experimental/ext_all.h)
\ No newline at end of file
+            ${PADDLE_INFERENCE_INSTALL_DIR}/paddle/include/experimental/ext_all.h)
```
Comment (chenwhql) [inline cmake/pten.cmake:43]: What changed here

File: cmake/pten.cmake (hunk 1 above)
Comment (kuizhiqing) [reply cmake/pten.cmake:24]: find file end with .h only

File: cmake/pten.cmake (hunk 2 above)
Comment (kuizhiqing) [reply cmake/pten.cmake:43]: nothing, pre commit did

**Pull request closed without merging**